lib/SILOptimizer/Transforms/DeadCodeElimination.cpp

```diff
@@ -258,6 +258,19 @@ void DCE::markLive(SILFunction &F) {
         }
         continue;
       }
+      if (isa<EndAccessInst>(&I)) {
+        if (auto *access = dyn_cast<BeginAccessInst>(I.getOperand(0))) {
```

Comment:
The operand of an end_access must be a begin_access. You can use EndAccessInst::getBeginAccess() to get it.

Reply:
Good to know. My thinking was that there could be a phi argument or something. Is that not valid?

Reply:
No, it must be a begin_access